src/DynamoWPFCLI/GeometryHolder.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Dynamo.Models;
+using Dynamo.Graph.Nodes;
+using Dynamo.Visualization;
+using Autodesk.DesignScript.Interfaces;
+
+namespace DynamoWPFCLI
+{
+    internal class GeometryHolder
+    {
+        private NodeModel node;
+        private ManualResetEvent done = new ManualResetEvent(false);
+        private Dictionary<string, Object> groupData = new Dictionary<string, object>();
+        private bool hasGeometry = false;
+
+        public Object Geometry
+        {
+            get
+            {
+                Done.WaitOne();
+                return groupData;
+            }
+        }
+
+        public ManualResetEvent Done
+        {
+            get { return done; }
+
+            private set { done = value; }
+        }
+
+        public bool HasGeometry
+        {
+            get
+            {
+                Done.WaitOne();
+                return hasGeometry;
+            }
+            private set { hasGeometry = value; }
+        }
+
+        public GeometryHolder(DynamoModel model, IRenderPackageFactory factory, NodeModel nodeModel)
+        {
+            node = nodeModel;
+            nodeModel.RenderPackagesUpdated += NodeRenderPackagesUpdated;
+            if (!nodeModel.RequestVisualUpdateAsync(model.Scheduler, model.EngineController, factory, true))
+            {
+                Done.Set();
+            }
+        }
+
+        public void Dispose()
+        {
+            node.RenderPackagesUpdated -= NodeRenderPackagesUpdated;
+        }
+
+        private void NodeRenderPackagesUpdated(NodeModel nodeModel, RenderPackageCache renderPackages)
```

Comment:
I guess I'm a little confused why this function can't just be a method that takes a list of renderPackages and serializes them. Is it to use JSON.net directly to make serialization more automatic?

Reply:
Well, this will change in the next sprint or two when we make our final decision of what format to use. The answer could be a direct serialization of the renderPackages or something else. I leave it "as is" for now because it happens to work with the testing tools I have.